### PR TITLE
EE-7886: Implement an agnostic way to inject diagnostic info for conn…

### PR DESCRIPTION
**Pull request by fsgonz**
…ectors

### CODE REVIEW

File: src/main/java/org/mule/extension/http/internal/request/HttpRequestOperations.java [hunk 1]
```diff
@@ -244,10 +254,37 @@ protected String buildPath(String basePath, String path) {
   public void initialise() throws InitialisationException {
     defaultStatusCodeValidator = new SuccessStatusCodeValidator("0..399");
     defaultRequestBuilder = new HttpRequesterRequestBuilder();
-    httpRequester = new HttpRequester(new HttpRequestFactory(), new HttpResponseToResult(), new HttpErrorMessageGenerator());
+    // Profiling API is only available for use in 4.5.0 or if enabled
+    httpResponseProfilingEnabled = runtimeVersion.atLeast("4.5.0-SNAPSHOT") || getBoolean(HTTP_ENABLE_PROFILING);
```
Comment (elrodro83) [inline src/main/java/org/mule/extension/http/internal/request/HttpRequestOperations.java:258]: do you really want this enabled by default on 4.5?

File: src/main/java/org/mule/extension/http/internal/request/HttpRequestOperations.java [hunk 2]
```diff
@@ -244,10 +254,37 @@ protected String buildPath(String basePath, String path) {
...
+    }
+
+    HttpProfilingServiceAdaptor profilingServiceAdaptor = new HttpProfilingServiceAdaptor();
+
+    // Manually inject the profiling service
+    try {
+      muleContext.getInjector().inject(profilingServiceAdaptor);
+    } catch (MuleException e) {
+      LOGGER.error("Exception attempting to obtain a profiling data producer", e);
+      return null;
```
Comment (elrodro83) [inline src/main/java/org/mule/extension/http/internal/request/HttpRequestOperations.java:282]: why not porpagate hie exception and fail?

File: src/main/java/org/mule/extension/http/internal/request/profiling/HttpProfilingServiceAdaptor.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Adaptor for {@link ProfilingService} to obtain a {@link HttpRequestResponseProfilingDataProducerAdaptor} for generating profiling related HTTP requests.
+ * This is implemented so that no need to change the min mule version is needed in the HTTP connector.
+ *
+ * @see HttpRequestOperations#initialise()
+ */
+public class HttpProfilingServiceAdaptor {
+
+  @Inject
+  ProfilingService profilingService;
```
Comment (elrodro83) [inline src/main/java/org/mule/extension/http/internal/request/profiling/HttpProfilingServiceAdaptor.java:25]: private

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/extension/http/internal/request/HttpRequestOperations.java (hunk 1 above)
Comment (fsgonz) [reply src/main/java/org/mule/extension/http/internal/request/HttpRequestOperations.java]: Available only by system property. Probably this should be changed in the future (best case feature flagging/toggling or activation strategy for extensions).

**Review by elrodro83 — APPROVED** [review]

**Comment by fsgonz** [comment]
--validate

**Pull request merged**